### PR TITLE
refactor: migrated feedback forms to svelte 5

### PR DESCRIPTION
**Pull request by gastoner**
### What does this PR do?
Migrates feedback forms to svelte5

### What issues does this PR fix or reference?
Part of #9192 

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/renderer/src/lib/feedback/SendFeedback.svelte [hunk 1]
```diff
@@ -29,7 +29,7 @@ function hideModal(): void {
 </script>
 
 {#if displayModal}
-  <Modal name="Share your feedback" on:close={hideModal}>
+  <Modal name="Share your feedback" onclose={hideModal}>
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/SendFeedback.svelte]: Not sure this is supported, Modal is our custom component.. not sure Svelte do the migration for us here

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 2]
```diff
@@ -16,16 +16,23 @@ import type { FeedbackProperties } from '/@api/feedback';
...
+let contactInformation = $state('');
+let hasFeedback = $state();
 
-$: hasFeedback =
-  (tellUsWhyFeedback && tellUsWhyFeedback.trim().length > 4) ||
-  (contactInformation && contactInformation.trim().length > 4);
+let { onCloseForm = () => {} }: Props = $props();
 
-export let onCloseForm = () => {};
+$effect(() => {
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte]: Issue: looks ks like it's more a $derived state than effect

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.spec.ts [hunk 3]
```diff
@@ -24,15 +24,19 @@ import { beforeAll, beforeEach, expect, test, vi } from 'vitest';
 
 import DevelopersFeedback from './DevelopersFeedback.svelte';
 
+const openExternalMock = vi.fn();
+const telemetryTrackMock = vi.fn();
+const sendFeedbackMock = vi.fn();
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.spec.ts]: Why declaring them on top? 

It should be avoid in favour of `vi.mocked(window.sendFeedback).mock....`

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.spec.ts [hunk 4]
```diff
@@ -24,15 +24,19 @@ import { beforeAll, beforeEach, expect, test, vi } from 'vitest';
...
+
 beforeAll(() => {
   Object.defineProperty(window, 'openExternal', {
-    value: vi.fn(),
+    value: openExternalMock,
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.spec.ts]: issue: AFAIK we don't need to have this change

vi.mocked(window.openExternal) gives you access to the mock

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 5]
```diff
@@ -3,23 +3,29 @@ import { Button, ErrorMessage, Link } from '@podman-desktop/ui-svelte';
...
-let issueTitle = '';
-let issueDescription = '';
-let issueValidaionError = '';
+interface Props {
+  onCloseForm: () => void;
+}
+
+let issueTitle = $state('');
+let issueDescription = $state('');
+let issueValidaionError = $state('');
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: maybe you can in a previous commit fix the name of this variable

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 6]
```diff
@@ -16,16 +16,23 @@ import type { FeedbackProperties } from '/@api/feedback';
...
+  hasFeedback =
+    (tellUsWhyFeedback && tellUsWhyFeedback.trim().length > 4) ||
+    (contactInformation && contactInformation.trim().length > 4);
+});
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte]: I think this effect could be placed inside a derived by state :thinking:

**Review by axel7083 — COMMENTED** [review]
Thanks for taking the time to do the migration :+1: really appreciate 😃

I noted some small changes

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 7]
```diff
@@ -16,16 +16,20 @@ import type { FeedbackProperties } from '/@api/feedback';
...
+let tellUsWhyFeedback = $state('');
+let contactInformation = $state('');
+let hasFeedback = $derived(
   (tellUsWhyFeedback && tellUsWhyFeedback.trim().length > 4) ||
-  (contactInformation && contactInformation.trim().length > 4);
+    (contactInformation && contactInformation.trim().length > 4),
+);
 
-export let onCloseForm = () => {};
+let { onCloseForm = () => {} }: Props = $props();
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte:32]: In the Props interface onCloseForm is not optional, the default value is not needed.

You may keep the default value and change the Props to something like 

```ts
interface Props {
  onCloseForm?: () => void;
}
```

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 8]
```diff
@@ -3,23 +3,29 @@ import { Button, ErrorMessage, Link } from '@podman-desktop/ui-svelte';
...
-let issueValidaionError = '';
+interface Props {
+  onCloseForm: () => void;
+}
+
+let issueTitle = $state('');
+let issueDescription = $state('');
+let issueValidationError = $state('');
+
+let { onCloseForm = () => {} }: Props = $props();
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: same remark, default not needed if onCloseForm is mandatory

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 9]
```diff
@@ -3,23 +3,29 @@ import { Button, ErrorMessage, Link } from '@podman-desktop/ui-svelte';
...
 
-$: if (!issueTitle) {
-  if (!issueDescription) {
-    issueValidaionError = 'Please enter bug title and description';
+$effect(() => {
```
Comment (axel7083) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: See https://svelte.dev/docs/svelte/$derived#$derived.by

**Comment by gastoner** [comment]
@axel7083 switched to derived and removed default value

**Review by axel7083 — APPROVED** [review]
LGTM

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 10]
```diff
@@ -16,16 +16,20 @@ import type { FeedbackProperties } from '/@api/feedback';
...
+let tellUsWhyFeedback = $state('');
+let contactInformation = $state('');
+let hasFeedback = $derived(
   (tellUsWhyFeedback && tellUsWhyFeedback.trim().length > 4) ||
-  (contactInformation && contactInformation.trim().length > 4);
+    (contactInformation && contactInformation.trim().length > 4),
+);
 
-export let onCloseForm = () => {};
+let { onCloseForm }: Props = $props();
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte]: I think it needs to have a default value if not specified

```suggestion
let { onCloseForm = () => {} }: Props = $props();
```

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 11]
```diff
@@ -3,26 +3,31 @@ import { Button, ErrorMessage, Link } from '@podman-desktop/ui-svelte';
...
+    if (!issueDescription) {
+      return 'Please enter bug description';
+    }
+    return '';
   }
-}
+});
 
-export let onCloseForm = () => {};
+let { onCloseForm }: Props = $props();
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: same remark there

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 12]
```diff
@@ -47,11 +52,11 @@ async function openGitHubIssues(): Promise<void> {
...
     {#if !issueTitle || !issueDescription}
-      <ErrorMessage class="text-xs" error={issueValidaionError}/>
+      <ErrorMessage class="text-xs" error={issueValidationError}/>
     {/if}
   </svelte:fragment>
   <svelte:fragment slot="buttons">
-    <Button class="underline" type="link" aria-label="Cancel" on:click={() => onCloseForm()}>Cancel</Button>
-    <Button aria-label="Preview on GitHub" on:click={() => openGitHubIssues()} disabled={!issueTitle || !issueDescription}>Preview on GitHub</Button>
+    <Button class="underline" type="link" aria-label="Cancel" onclick={() => onCloseForm?.()}>Cancel</Button>
+    <Button aria-label="Preview on GitHub" onclick={() => openGitHubIssues()} disabled={!issueTitle || !issueDescription}>Preview on GitHub</Button>
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: AFAIK you can't change on:click to onclick as the component is still expecting an event, not a callback

**Review by benoitf — CHANGES_REQUESTED** [review]
it does not work

you can't click on the buttons

**Comment by benoitf** [comment]
https://github.com/user-attachments/assets/2992b713-a398-4924-96f6-8853c1eb479f

**Comment by gastoner** [comment]
@axel7083 I've rebased due to some conflicts. Can you check again? Thanks 🙏

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 13]
```diff
@@ -48,11 +52,11 @@ async function sendFeedback(): Promise<void> {
   // send
   await window.sendFeedback(properties);
   // close the form
-  onCloseForm();
+  onCloseForm?.();
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte]: revert
```suggestion
  onCloseForm();
```

File: packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte [hunk 14]
```diff
@@ -48,11 +52,11 @@ async function sendFeedback(): Promise<void> {
...
 }
 
 async function openGitHub(): Promise<void> {
-  onCloseForm();
+  onCloseForm?.();
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/DevelopersFeedback.svelte]: revert
```suggestion
  onCloseForm();
```

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 15]
```diff
@@ -4,35 +4,40 @@ import { Button, ErrorMessage, Link } from '@podman-desktop/ui-svelte';
...
+let descriptionPlaceholder = $state(category === 'bug' ? 'Bug description' : 'Feature description');
+let existingIssuesLink = $state(
+  category === 'bug'
+    ? 'https://github.com/podman-desktop/podman-desktop/issues?q=label%3A%22kind%2Fbug%20%F0%9F%90%9E%22'
+    : 'https://github.com/podman-desktop/podman-desktop/issues?q=label%3A%22kind%2Ffeature%20%F0%9F%92%A1%22',
+);
 
 async function openGitHubIssues(): Promise<void> {
-  onCloseForm();
+  onCloseForm?.();
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: revert
```suggestion
  onCloseForm();
```

File: packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte [hunk 16]
```diff
@@ -57,11 +62,11 @@ async function openGitHubIssues(): Promise<void> {
   </svelte:fragment>
   <svelte:fragment slot="validation">
     {#if !issueTitle || !issueDescription}
-      <ErrorMessage class="text-xs" error={issueValidaionError}/>
+      <ErrorMessage class="text-xs" error={issueValidationError}/>
     {/if}
   </svelte:fragment>
   <svelte:fragment slot="buttons">
-    <Button class="underline" type="link" aria-label="Cancel" on:click={() => onCloseForm()}>Cancel</Button>
+    <Button class="underline" type="link" aria-label="Cancel" on:click={() => onCloseForm?.()}>Cancel</Button>
```
Comment (benoitf) [inline packages/renderer/src/lib/feedback/feedbackForms/GitHubIssueFeedback.svelte]: revert
```suggestion
    <Button class="underline" type="link" aria-label="Cancel" on:click={() => onCloseForm()}>Cancel</Button>
```

**Review by benoitf — APPROVED** [review]
please revert the .?() call on onCloseForm before merging

**Review by axel7083 — APPROVED** [review]
LGTM

**Pull request merged**